enkit/outputs/commands.go

```diff
@@ -0,0 +1,187 @@
+		Command: &cobra.Command{
+			Use:   "unmount [invocation ID]",
+			Short: "Unmount the build outputs of a particular invocation",
+			Example: `  $ enkit outputs unmount 73d4a9f0-a0c4-4cb2-80eb-b4b4b9720d07
+	Unmounts outputs from build 73d4a9f0-a0c4-4cb2-80eb-b4b4b9720d07 from the
+	default location.`,
+			Aliases: []string{"umount"},
+			Args:    cobra.ExactArgs(1),
+		},
+		root: root,
+	}
+	command.Command.RunE = command.Run
+	return command
+}
+
+func (c *Unmount) Run(cmd *cobra.Command, args []string) error {
+	return fmt.Errorf("`enkit outputs unmount` is unimplemented")
+}
+
+type Run struct {
+	*cobra.Command
+	root *Root
+
+	InvocationID string
+	DirPath      string
+	ZipPath      string
+}
+
+func NewRun(root *Root) *Run {
+	command := &Run{
+		Command: &cobra.Command{
+			Use:   "run",
+			Short: "Mount build artifacts and execute an optional command on them",
+			Example: `  $ enkit outputs run --invocation=73d4a9f0-a0c4-4cb2-80eb-b4b4b9720d07
+	Launches a shell with build outputs from a particular build re-rooted so that
+	paths are correct within the outputs themselves.
+
+  $ enkit outputs run --dir=/tmp/some_dir
+	Launches a shell with build outputs in /tmp/some_dir re-rooted so that paths
+	are correct within the outputs themselves.
+
+  $ enkit outputs run --zip=/tmp/some.zip
+	Unpacks the named zip into a tempdir, and then reroots the artifacts so that
+	paths are correct within the outputs themselves.
+
+  $ enkit outputs run --zip=/tmp/some-zip -- find .
+	Runs "find ." in the unpacked zip after it is rerooted.`,
+		},
+		root: root,
+	}
+
+	command.Command.RunE = command.Run
+	command.Command.PreRunE = command.validate
+	command.Flags().StringVar(&command.InvocationID, "invocation_id", "", "If set, the build's invocation ID from which to mount artifacts")
+	command.Flags().StringVar(&command.DirPath, "dir_path", "", "If set, the path to already-unpacked artifacts to reroot")
+	command.Flags().StringVar(&command.ZipPath, "zip_path", "", "If set, the path to a zipped outputs file to unpack and re-root")
+	return command
+}
+
+func (c *Run) Run(cmd *cobra.Command, args []string) error {
+	return fmt.Errorf("`enkit outputs run` is unimplemented")
+}
+
+func (c *Run) validate(cmd *cobra.Command, args []string) error {
+	setCount := 0
+	for _, f := range []string{c.InvocationID, c.DirPath, c.ZipPath} {
+		if f != "" {
+			setCount++
+		}
+	}
+	switch {
+	case setCount == 0:
+		return fmt.Errorf("One of --invocation_id, --dir_path, --zip_path must be set")
+	case setCount >= 2:
+		return fmt.Errorf("Only one of --invocation_id, --dir_path, --zip_path may be set")
+	}
+	return nil
+}
+
+type Shutdown struct {
+	*cobra.Command
+	root *Root
+}
+
+func NewShutdown(root *Root) *Shutdown {
+	command := &Shutdown{
+		Command: &cobra.Command{
+			Use:   "shutdown",
+			Short: "Unmount all builds under particular directory",
+			Example: `  $ enkit outputs shutdown
+	Unmounts bb_clientd, and deletes all links to build artifacts.`,
```

Comment:
remove references to bb_clientd, we are using bb_clientd now as implementation, but our average developer may not know / will likely not have direct visibility into it?

Reply:
Agreed; I'll replace this with something less specific.